paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/RemoveUnexistingFilesAction.java

```diff
@@ -0,0 +1,449 @@
+                                "List Buckets",
+                                BasicTypeInfo.STRING_TYPE_INFO,
+                                new ListBucketsInPartitionOperator(fileStoreTable))
+                        .transform(
+                                "List Files",
+                                BasicTypeInfo.STRING_TYPE_INFO,
+                                new ListFilesInBucketOperator(fileStoreTable.fileIO()));
+        SingleOutputStreamOperator<Tuple4<byte[], Integer, byte[], String>> splits =
+                source.transform(
+                        "List Data Splits",
+                        new TupleTypeInfo<>(
+                                PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO,
+                                BasicTypeInfo.INT_TYPE_INFO,
+                                PrimitiveArrayTypeInfo.BYTE_PRIMITIVE_ARRAY_TYPE_INFO,
+                                BasicTypeInfo.STRING_TYPE_INFO),
+                        new ScanManifestOperator(fileStoreTable));
+
+        SingleOutputStreamOperator<Committable> comparator =
+                files.keyBy(f -> f)
+                        .connect(splits.keyBy(t -> new Path(t.f3).getName()))
+                        .transform(
+                                "Compare Files",
+                                new CommittableTypeInfo(),
+                                new CompareOperator(fileStoreTable));
+        DataStream<String> result = comparator.getSideOutput(RESULT_SIDE_OUTPUT);
+        if (dryRun) {
+            return result;
+        }
+
+        comparator
+                .transform(
+                        "Global Committer : " + table.name(),
+                        new CommittableTypeInfo(),
+                        new CommitOperator(fileStoreTable))
+                .forceNonParallel();
+        return result;
+    }
+
+    @Override
+    public void run() throws Exception {
+        build();
+        env.execute("Remove Unexisting Files : " + table.name());
+    }
+
+    private static class ListBucketsInPartitionOperator
+            extends BoundedOneInputOperator<byte[], String> {
+
+        private static final long serialVersionUID = 1L;
+
+        private final FileStoreTable table;
+
+        private transient BinaryRow reuse;
+        private transient FileStorePathFactory pathFactory;
+
+        private ListBucketsInPartitionOperator(FileStoreTable table) {
+            this.table = table;
+        }
+
+        @Override
+        public void open() throws Exception {
+            reuse = new BinaryRow(table.schema().partitionKeys().size());
+            pathFactory = table.store().pathFactory();
+        }
+
+        @Override
+        public void processElement(StreamRecord<byte[]> record) throws Exception {
+            byte[] bytes = record.getValue();
+            reuse.pointTo(MemorySegment.wrap(bytes), 0, bytes.length);
+            Path partitionPath = pathFactory.getPartitionPath(reuse);
```

Comment:
We need to consider external path for table...

I think it is better to still use fileio.exists method...